src/components/main/MoveTreeModal.tsx

```diff
@@ -0,0 +1,259 @@
+      (ele: DiscriminatedItem) => ele.type === ItemType.FOLDER,
+    );
+  }, [selectSubItems]);
+  return (
+    <div id={`${TREE_MODAL_MY_ITEMS_ID}`}>
+      {/* Home or Root  Which will be the start of the menu */}
+      {isHome && (
+        <MoveMenuRow
+          key={rootMenuItem.name}
+          ele={rootMenuItem}
+          fetchSubItems={() => {
+            setIsHome(false);
+            setPaths((paths: DiscriminatedItem[]) => [...paths, rootMenuItem]);
+            setSelectedId(rootMenuItem.id);
+          }}
+          selectedId={selectedId}
+          setSelectedId={setSelectedId}
+          itemIds={[]}
+        />
+      )}
+      {/* end of home */}
+
+      {/* Items for selected Item, So if I choose folder1 this should be it children */}
+      {!isHome &&
+        items?.map((ele) => (
+          <MoveMenuRow
+            key={ele.id}
+            ele={ele}
+            fetchSubItems={() => selectSubItems(ele)}
+            selectedId={selectedId}
+            setSelectedId={setSelectedId}
+            itemIds={itemIds}
+          />
+        ))}
+
+      {/* Default Parent So If I want to move an item who's in nested folder we will have it's parent as default */}
+
+      {defaultParent && !selectedSubItem && (
+        <MoveMenuRow
+          key={defaultParent.id}
+          ele={defaultParent}
+          fetchSubItems={() => {
+            selectSubItems(defaultParent);
+            setIsHome(false);
+            setSelectedId(defaultParent.id);
+          }}
+          selectedId={selectedId}
+          setSelectedId={setSelectedId}
+          itemIds={itemIds}
+        />
+      )}
+
+      {!isHome && !defaultParent && !items?.length && (
+        <div>
+          {translateBuilder(BUILDER.EMPTY_FOLDER_CHILDREN_FOR_THIS_ITEM)}
+        </div>
+      )}
+    </div>
+  );
+};
+
+const TreeModal = ({
+  title,
+  onClose,
+  onConfirm,
+  open = false,
+  itemIds = [],
+}: TreeModalProps): JSX.Element => {
+  const { t: translateBuilder } = useBuilderTranslation();
+
+  const [selectedId, setSelectedId] = useState<string>('');
+  // serious of breadcrumbs
+  const [paths, setPaths] = useState<DiscriminatedItem[]>([]);
+
+  const { data: parentItem } = hooks.useItem(
+    selectedId === TREE_MODAL_MY_ITEMS_ID ? '' : selectedId,
+  );
+  const { data: parents } = hooks.useParents({
+    id: itemIds?.[0],
+  });
+  const [defaultSelectedSubItem, setDefaultSelectedSubItem] =
+    // eslint-disable-next-line no-unsafe-optional-chaining
+    useState<DiscriminatedItem | null>(null);
+
+  const handleClose = () => {
+    onClose({ id: null, open: false });
+  };
+
+  const onClickConfirm = () => {
+    onConfirm({ ids: itemIds, to: selectedId });
+    handleClose();
+  };
+
+  return (
+    <Dialog
+      onClose={handleClose}
+      aria-labelledby={dialogId}
+      open={open}
+      scroll="paper"
+    >
+      <DialogTitle id={dialogId}>{title}</DialogTitle>
+      <DialogContent sx={{ height: '250px' }}>
```

Comment:
Let's not fix the height. I notice you have the flickering with the width as well anyway.